uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/models/instances/leave-request-instance_test.js

```diff
@@ -1,137 +1,284 @@
-            });
+      beforeEach(function () {
+        commonSetup(mockData.singleDataError());
+      });
+
+      it('updates the status_id of the instance', function () {
+        promise.then(function (data) {
+          expect(data).toBe(mockUpdateResponse);
         });
+      });
+    })
+  });
+
+  describe('update()', function () {
+    var instanceUpdate, newRequestData;
+
+    beforeEach(function () {
+      requestData = mockData.all().values[0];
+      instance = LeaveRequestInstance.init(requestData, true);
+      var changedStatusId = {
+        status_id: mockData.all().values[5].status_id
+      };
+      newRequestData = _.assign({}, requestData, changedStatusId);
+    });
+
+    describe('when id is set', function () {
+
+      beforeEach(function () {
+        _.assign(instance, newRequestData);
+        instanceUpdate = instance.update();
+      });
+
+      afterEach(function () {
+        //to excute the promise force an digest
+        $rootScope.$apply();
+      });
 
-        it('LeaveRequestInstance.update gets called', function () {
-          promise.then(function () {
-              expect(LeaveRequestInstance.update).toHaveBeenCalledWith({
-                'status_id': mockOptionValue[0].value
-              });
-            });
+      it('calls equivalent API method', function () {
+        instanceUpdate.then(function () {
+          expect(LeaveRequestAPI.update).toHaveBeenCalled();
         });
       });
 
-      describe("error", function () {
+      it('modifies attributes', function () {
+        var updatedAttributes = _.assign(Object.create(null), requestData, newRequestData);
 
-        beforeEach(function () {
-          commonSetup(mockData.singleDataError());
+        instanceUpdate.then(function () {
+          expect(instance.attributes).toEqual(jasmine.objectContaining(updatedAttributes));
         });
+      });
+    });
 
-        it('updates the status_id of the instance', function () {
-          promise.then(function (data) {
-            expect(data).toBe(mockUpdateResponse);
-          });
+    describe('when id is missing', function () {
+
+      beforeEach(function () {
+        expectedError = {
+          is_error: 1,
+          error_message: 'id is mandatory field'
+        };
+        delete newRequestData.id;
+        _.assign(instance, newRequestData);
+        instanceUpdate = instance.update();
+      });
+
+      afterEach(function () {
+        //to excute the promise force an digest
+        $rootScope.$apply();
+      });
+
+      it('fails to update attributes ', function () {
+        instanceUpdate.catch(function (error) {
+          expect(error).toEqual(jasmine.objectContaining(expectedError));
         });
-      })
+      });
+    });
+  });
+
+  describe('create()', function () {
+    var instanceCreate;
+
+    beforeEach(function () {
+      requestData = helper.createRandomLeaveRequest();
+      instance = LeaveRequestInstance.init(requestData, true);
```

Comment:
When you create any instance, the data does not come from the api (since you're creating it for the first time), so you shouldn't pass `true` to `init()`